plugins/modules/mso_schema_template_anp_epg_selector.py

```diff
@@ -0,0 +1,266 @@
+    if anp not in anps:
+        mso.fail_json(msg="Provided anp '{anp}' does not exist. Existing anps: {anps}".format(anp=anp, anps=', '.join(anps)))
+    anp_idx = anps.index(anp)
+
+    # Get EPG
+    epgs = [e.get('name') for e in schema_obj.get('templates')[template_idx]['anps'][anp_idx]['epgs']]
+    if epg not in epgs:
+        mso.fail_json(msg="Provided epg '{epg}' does not exist. Existing epgs: {epgs}".format(epg=epg, epgs=', '.join(epgs)))
+    epg_idx = epgs.index(epg)
+
+    # Get Selector
+    selectors = [s.get('name') for s in schema_obj.get('templates')[template_idx]['anps'][anp_idx]['epgs'][epg_idx]['selectors']]
+    if selector in selectors:
+        selector_idx = selectors.index(selector)
+        selector_path = '/templates/{0}/anps/{1}/epgs/{2}/selectors/{3}'.format(template, anp, epg, selector_idx)
+        mso.existing = schema_obj.get('templates')[template_idx]['anps'][anp_idx]['epgs'][epg_idx]['selectors'][selector_idx]
+
+    if state == 'query':
+        if selector is None:
+            mso.existing = schema_obj.get('templates')[template_idx]['anps'][anp_idx]['epgs'][epg_idx]['selectors']
+        elif not mso.existing:
+            mso.fail_json(msg="Selector '{selector}' not found".format(selector=selector))
+        mso.exit_json()
+
+    selectors_path = '/templates/{0}/anps/{1}/epgs/{2}/selectors/-'.format(template, anp, epg)
+    ops = []
+
+    mso.previous = mso.existing
+    if state == 'absent':
+        mso.existing = {}
+        ops.append(dict(op='remove', path=selector_path))
+
+    elif state == 'present':
+        # Get expressions
+        all_expressions = []
+        if expressions:
+            for expression in expressions:
+                operator = expression.get('operator')
+                if operator in ["keyExist", "keyNotExist"] and expression.get("value"):
+                    mso.fail_json(
+                        msg="attribute 'value' is not supported for operator '{0}' in expression '{1}'".format(operator, expression.get('type')))
+                if operator in ["notIn", "in", "equals", "notEquals"] and not expression.get("value"):
+                    mso.fail_json(
+                        msg="attribute 'value' needed for operator '{0}' in expression '{1}'".format(operator, expression.get('type')))
+                all_expressions.append(dict(
+                    key='Custom:' + expression.get('type'),
+                    operator=operator,
+                    value=expression.get('value'),
+                ))
+
+        payload = dict(
+            name=selector,
+            expressions=all_expressions,
+        )
+
+        mso.sanitize(payload, collate=True)
+
+        if mso.existing:
+            ops.append(dict(op='replace', path=selector_path, value=mso.sent))
+        else:
+            ops.append(dict(op='add', path=selectors_path, value=mso.sent))
+
+        mso.existing = mso.proposed
+
+    if not module.check_mode:
```

Comment:
I think this condition should be : "if not module.check_mode and mso.proposed != mso.previous:" to avoid unnecessary API calls.